QueryField: Do not modify the query field while typing

ifrost opened the pull request:
**What this PR does / why we need it**:

[This PR](https://github.com/grafana/grafana/pull/34076) introduced a bug in QueryField - the input may lose focus while typing. The problem seems to stem from the value being sanitized while typing. This is triggering an update in Slate component causing the blur and eventually losing focus on the field [here](https://github.com/grafana/slate/blob/master/packages/slate-react/src/components/content.js#L233).

It's also somehow related to debouncing the update by 500ms. The way to reproduce it is to write any text and then after the last letter quickly type a single space before 500ms elapses. (`cleanText` method trims the value removing that space on change)

![blur](https://user-images.githubusercontent.com/745532/118670235-b74c4980-b7f6-11eb-95d0-406516f5a1df.gif)

The cursor will disappear after the space is typed and 500ms debounce finishes.

~This PR changes the logic so cleaning the text doesn't happen while typing but only onBlur and before running the query. Not sure if that's the best approach. Also I'm not 100% sure why it happens only if the space is typed quickly after the last letter, but I can try to check this.~

Other approach could also be get rid of trimming, but not sure if we can do it. 

Update: as discussed - ^^this is the current solution.

<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

-->

**Which issue(s) this PR fixes**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes N/A

**Special notes for your reviewer**: N/A

ifrost committed:
Do not sanitize the query while typing, but only onBlur or before the query is run

Otherwise sanitized value may cause update Slate component and trigger blur losing focus of the input

ivanahuckova reviewed (commented):
>Other approach could also be get rid of trimming, but not sure if we can do it.

I think I like the idea of removing trimming better. I personally think that it is simpler and cleaner solution. The trimming was just additionally added at the end and the fix works also without it.

ifrost committed:
Revert "Do not sanitize the query while typing, but only onBlur or before the query is run"

This reverts commit 00779889

ifrost committed:
Avoid changing the input while typing

Cleaning happens on each change and with trimming it changes the value if the user types a space as the last character (may happen quite often while typing a query). In worst cases it's causing losing the focus in Slate if the space is typed before debounced change callback is triggered (500ms).

ifrost commented:
> > Other approach could also be get rid of trimming, but not sure if we can do it.
> 
> I think I like the idea of removing trimming better. I personally think that it is simpler and cleaner solution. The trimming was just additionally added at the end and the fix works also without it.

Removed trimming 💇‍♂️
